### DISCUSSION TITLE
[next] Spec for automatic router merge/organization via folder

### DISCUSSION THREAD

**Discussion by lkj4** [trpc/trpc#2042] [Ideas]
@KATT, not sure where I should put this issue, the Github Project for `next` feels a bit dead, feel free to move this issue.

My preferred way for router organization would be throwing all my routers into one folder and each file in that folder just default exports `t.router({...` without the need of a manual `t.mergeRouter(...`.

The earlier we have a spec we can already organize our routers according to this spec and once the feature is in place we just need to add one line to get it working without a big migration hassle.

**Comment by sachinraja** [comment]
Thanks for the idea! `next` is actively being worked on, the project hasn't been updated simply because...it's being worked on. See PRs:
- https://github.com/trpc/trpc/pull/2024
- https://github.com/trpc/trpc/pull/2032

As for the folder organization, I don't really see how that would work with types.

**Comment by KATT** [comment]
Simply remarking that the development of something is slow without any real offering to help out is a bit provoking, I really don't see what you want to achieve here? 

tRPC in its current form already has unparalleled DX, there are few immediate issues and any API changes needs to be well-considered. If you're not looking to actively help out, just please back off and have patience instead. 

We're not really getting paid much for this and are all still working every day to push the project moving forward. You are very welcome to join in the efforts though, feel free to @ me on github or whatnot.

**Reply by KATT** [reply]
Maybe this came off a bit unnecessarily abrasive, apologies for that. I do appreciate input and I can assure you that we're working very hard on the `next` branch. I even moved a lot of tickets on that project around the other day ☺

**Reply by lkj4** [reply]
> without any real offering to help out is a bit provoking,

Was my tone so off that I provoked you?

> If you're not looking to actively help out, just please back off and have patience instead.

Alex, I _just_ brought up an idea, why do you think I am not looking to actively help out?? Why are you so angry? 

And btw, you should welcome every user who finds bugs, shares ideas or whatever, all this will help you growing tRPC.

Shutting down users ("back off") because you don't like the tone? I've even promoted tRPC v10 in SvelteKit's GitHub Discussion.

> tRPC in its current form already has unparalleled DX,

Yeah, the DX is quite nice, especially with v10,. But it's still a long way, it's not just about missing stuff here and there but you need to get a community around tRPC and users need to be sure that the is repo won't get unmaintained in a year or so. Then, you need to get the same traction in side repos such as the SvelteKit one and so on.

I am sure you get there but I am still baffled about your reaction...

BTW, what I (and probably others) would like to know why did Colin leave the project?

**Reply by lkj4** [reply]
Alex, sorry haven't read you last short message! thanks for the kind words and can't take back my prior words anymore. Great project and best rpc lib I've ever used, keep on! 🙂🙂

**Reply by KATT** [reply]
I've gotten a few comments that things aren't moving fast enough so I got a bit triggered with the "seems dead"-comment when I've been spending so many nights and weekends over the last few months on it. It's free software and yet people, who never contributed a single commit nor even sponsor the project with a fiver a month, feel entitled enough to complain.

All love, thanks for your kind words as well!
